nistats/_utils/datasets.py

```diff
@@ -0,0 +1,93 @@
+import json
+import os
+
+from nilearn.datasets.utils import _get_dataset_dir
+
+
+def make_openneuro_dataset_urls_index(
+        data_dir=None,
+        dataset_version='ds000030_R1.0.4',
+        verbose=1,
+        ):
+    """Creates a fresh, updated OpenNeuro BIDS dataset index from AWS,
+    ready for upload to osf.io
+
+    Currently, the fetcher downloads the index (a sorted list of urls),
+    using :func:`datasets.fetch_openneuro_dataset_index` from osf.io ,
+    since this approach is faster and circumvents `boto` as a dependency.
+
+    If the URLs list become outdated in future, this function
+    creates a file 'nistats_fetcher_openneuro_dataset_urls.json'
+    with an updated list of URLs, ready to be uploaded as quick-files
+    to Nilearn account on osf.io.
+    To ensure that the same download link is updated,
+    the file name must remain the same.
+
+    Note: This function requires ``boto3`` to be installed.
```

Comment:
There is no way to avoid this ?

Reply:
No. This function updates the URL index we have uploaded to osf.io, in case OpenNeuro changes the layout .
Boto3 is used in this function (not using it is too much work for almost no benefit). But it is no longer a user facing dependency.